### ISSUE TITLE
Test with Janus 1.x

### ISSUE THREAD

**Issue by fancycode** [strukturag/nextcloud-spreed-signaling#280]
There seem to be problems when running against Janux 1.x, see https://github.com/strukturag/nextcloud-spreed-signaling/issues/279.

**Comment by fancycode** [comment]
Not a problem with 1.x according to #279, still should be tested though.

**Comment by brknkfr** [comment]
I just tested nextcloud-spreed-signaling on a newly installed Debian system (Bullseye) with janus version 1.x (actually 1.0.1-1~bpo11+1 from the bullseye-backports) and it seems to work just fine.

**Comment by muchahagrande** [comment]
I've been using 1.x versions of Janus for some time and I can confirm that is working as with 0.x versions.
I'm using this docker to run Janus: mwalbeck/janus-gateway. It is always updated to the last Meetecho version.

**Comment by muchachagrande** [comment]
Hi, I've just fallen into a problem related with Talk for Android. Since then (version 15.1.1 and beyond) there were no video calls on when using signaling server.
Yesterday I took the time to try to debug the problem and I found that it was a Janus version problem.
Since version 1.x Janus supports multi stream peer connections and it works with some differences than the 0.x versions that are still maintained in parallel.
The problem is described on this issue https://github.com/meetecho/janus-gateway/issues/2566
Nextcloud Talk expects that the data channel to receive control messages from is labeled "JanusDataChannel", but in the multistream version they must include the participant ID in the label in order to identify who the data comes from.
So, with the multistream version the data channel is not JanusDataChannel anymore causing that Talk complains about receiving "Unknown message from 1 datachannel". The "1" is the label of the data channel and is not a known label to Talk.
Now that I've tested with a Janus version 0.x I can confirm that there are no more Unknown Messages received and Talk Android works again.

**Comment by wargreen** [comment]
Hi,
I've just setup  the signaling server for our NC instance.
Janus 0.x don't seem to be anymore in the debian's repos.
Is it any news about the issue ? Or any workaround on debian-based distro ?

Thank you

**Comment by fancycode** [comment]
Looks like no changes are necessary to support Janus 1.x in the signaling server itself.

See the following Talk-related issues for changes necessary in clients using the signaling server / Janus:
- Web: https://github.com/nextcloud/spreed/issues/10003
- Android: https://github.com/nextcloud/talk-android/issues/2704
- iOS: https://github.com/nextcloud/talk-ios/pull/1557

**Comment by mircokam** [comment]
Hello,
we testet the Android App Talk Version 18.1.0 on Pixel 7 with Janus 1.x Multistream  an Nextcloud 28.0.3 and talk 18.0.5 an it works now with video. 

https://github.com/sunweaver/nextcloud-high-performance-backend-setup/issues/109

**Comment by muchachagrande** [comment]
Did you test a video call from android to android ?

**Comment by mircokam** [comment]
I tested now android to android and at first, the camera streams aren't transmitted between the android devices (they are transmitted on/to iOS and in/to the Web). To transmit them, you have to disable and reenable the camera on both android devices in the video call. After that it works fine. I'm gonna open a new issue in the repo of the android app.

**Comment by muchachagrande** [comment]
Ok... I was wondering if the upgrade of HPB would solve that issue, but it is not the case... thank you

**Comment by fancycode** [comment]
> Ok... I was wondering if the upgrade of HPB would solve that issue, but it is not the case... thank you

All these issues are purely client side, see https://github.com/strukturag/nextcloud-spreed-signaling/issues/280#issuecomment-1966309538 for the PRs in the client apps to support Janux 1.x. If you still experience (other) problems, please open a new issue in the related app repository.

**Comment by muchachagrande** [comment]
Hi @fancycode.

> If you still experience (other) problems, please open a new issue in the related app repository.

There is an already opened issue for the referenced problem in the proper repository.

Thank you and cheers
